docs/ruleset_functions/get_primary_secondary_loops.md

```diff
@@ -0,0 +1,61 @@
+
+  - For each HVAC system in building segment: `for hvac in building_segment.heating_ventilation_air_conditioning_systems`
+
+    - Check if HVAC system has chilled water coil: `if hvac.cooling_system.chilled_water_loop:`
+
+      - Save chilled water loop that serves cooling systems to CHW coil loop array to get all loops connected to cooling coils: `chw_coil_loop_array.append(hvac.cooling_system.chilled_water_loop)` (Note XC, need to include zonal cooling coil as well once the schema is updated)
+
+- For each fluid loop in RMR: `for fluid_loop in RMR.RulesetModelInstance.fluid_loops:`
+
+  - Check if loop type is the same as input loop_type: `if fluid_loop.type == loop_type:`
+
+    - Check if loop is connected to chiller(s): `if fluid_loop.id in component_loop_array:`
+
+      - Check if loop has child loop(s) and loop is not connected to any cooling coils: `if ( fluid_loop.child_loops != NULL ) AND ( NOT fluid_loop.id in chw_coil_loop_array ):`
+
+        - Set pass primary check flag to True: `pass_primary_check_flag = TRUE`
+
+        - Save loop to primary loop array: `primary_loop_array.append(fluid_loop.id)`
+
+        - Save all child loops to child loop array: `child_loop_array.append(child_loop_id for child_loop_id in fluid_loop.child_loops)`
+
+    - Check if loop to connected to cooling coil(s): `if fluid_loop in chw_coil_loop_array:`
+
+      - Check if loop does not have any child loop and is not connected to any chiller(s): `if ( fluid_loop.child_loops == NULL ) AND ( NOT fluid_loop.id in component_loop_array ):`
+
+        - Set pass secondary check flag to True: `pass_secondary_check_flag = TRUE`
+
+        - Save loop to secondary loop array: `secondary_loop_array.append(fluid_loop.id)`
+
+- Check if child loop array and secondary loop array is the same, set pass child loop check flag to True: `if child_loop_array.sort() == secondary_loop_array.sort(): pass_child_loop_check_flag = TRUE`
+
+- If all three flags are true, save primary and secondary loop to output dictionary: `if pass_primary_check_flag AND pass_secondary_check_flag AND pass_child_loop_check_flag: primary_secondary_loop_dictionary["PRIMARY"] = primary_loop_array, primary_secondary_loop_dictionary["SECONDARY"] = secondary_loop_array`
+
+- Else, primary secondary configuration is not modeled correctly, save empty list to output dictionary: `else: primary_secondary_loop_dictionary["PRIMARY"] = [], primary_secondary_loop_dictionary["SECONDARY"] = []`
+
+**Returns** `return primary_secondary_loop_dictionary`  
+
+**Notes:**
+
+1. Primary secondary configuration is correct if:
+1). a cooling loop is either connected to chiller or cooling coil, but not both
+2). if a cooling loop is connected to chiller, it shall have child loop(s)
+3). if a cooling loop is connected to cooling coil, it shall not have any child loops
+4). the collection of child loops of all of 2) shall be the same as all of 3)
```

Comment:
@xingcx as discussed, maybe we can add some information here to help future reviewers
1. Primary Only Loop: A cooling loop which is connected to chiller and cooling coil.
2. Primary Secondary Loop: Primary loop would be connected to chiller but not to cooling coil. The loop connected to the chiller would have child loop(s). The secondary loop would be connected to the cooling coil and not have child loop.

Reply:
as discussed on 2022.04.26, we discussed to modify the function to return the list of primary loops and the list of secondary rules

Reply:
As discussed, 

- 22-25, would verify the primary loop is continuous constant flow
- 22-36, would verify the secondary loop is variable
- Need one more RDS to verify that baseline loop configuration is primary secondary

Reply:
Pump rules:
22-25 would verify the primary pump is constant speed.
22-10 would verify the secondary pump is constant speed (<300ton) or with VFD (>=300ton).
Loop rules:
22-7 would verify that baseline has been modeled with P S configuration.
22-36 would verify continuous constant flow primary loop and variable-flow secondary loop.
22-34 would verify baseline has been modeled with only one CHW plant (i.e. one primary and one secondary loop).